crates/transaction-pool/src/lib.rs

```diff
@@ -473,6 +473,18 @@ where
         self.pool.get_transactions_by_sender(sender)
     }
 
+    fn get_transactions_by_sender_and_nonce(
+        &self,
+        sender: Address,
+        nonce: u64,
+    ) -> Vec<Arc<ValidPoolTransaction<Self::Transaction>>> {
```

Comment:
(address, nonce) pairs are unique in the pool, so that there is always only one matching tx

Reply:
I realized I wasn't clear on that in the issue, sorry about that -.-

Reply:
No problem :) Just modified it, tell me if it sounds good to you now

Reply:
And this is my fault nonce is here to protect against replay attacks. So this is obvious that the nonce is unique for a given address. Just didn't think well :)